### ISSUE TITLE
Update to latest JSON-LD 1.1 lib

### ISSUE THREAD

**Issue by dmitrizagidulin** [digitalbazaar/jsonld-signatures#106]
Update to use the JSON-LD 1.1 version of `jsonld.js`. (Which will enable framing, other features.)

**Comment by dmitrizagidulin** [comment]
Upgrading to jsonld.js `v3.1.1` yields several instances of this error:

`TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))`

```
       Ed25519Signature2018
         should verify a document w/security context:

      AssertionError: expected { Object (verified, results, ...) } to deeply equal { Object (verified, results) }
      + expected - actual

       {
      -  "error": "VerificationError: Verification error(s)."
         "results": [
           {
      -      "error": "TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))"
             "proof": {
               "@context": "https://w3id.org/security/v2"
               "created": "2018-02-13T21:26:08Z"
               "creator": "https://example.com/i/carol/keys/1"
               "jws": "eyJhbGciOiJFZERTQSIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..UNcNI6x6KDA_hHux2RLM8_i9aoZY34GwcZevOjkSh22WoNB4FcP6dNgf2nKzXXJIr-IqUnEwMYeD36fc8jv1AA"
               "proofPurpose": "https://example.org/special-authentication"
               "type": "Ed25519Signature2018"
             }
      -      "verified": false
      +      "purposeResult": {
      +        "valid": true
      +      }
      +      "verified": true
           }
         ]
      -  "verified": false
      +  "verified": true
       }
      at Context.<anonymous> (tests/test-common.js:422:16)
```

**Comment by dmitrizagidulin** [comment]
Specifically, it's failing at `lib/suites/LinkedDataSignatures.js:252`

```
    // Note: `expansionMap` is intentionally not passed; we can safely drop
    // properties here and must allow for it
    const {'@graph': [framed]} = await jsonld.frame(verificationMethod, {
      '@context': constants.SECURITY_CONTEXT_URL,
      '@embed': '@always',
      id: verificationMethod
    }, {documentLoader, compactToRelative: false});
    if(!framed) {
      throw new Error(`Verification method ${verificationMethod} not found.`);
    }
```

**Comment by dlongley** [comment]
`@graph` is probably not present and therefore its value and `framed` are probably undefined. Try passing `omitGraph: false` as an option to the `jsonld.frame` call.

**Comment by dmitrizagidulin** [comment]
Didn't help, investigating further.

**Comment by SamHellawell** [comment]
I had the same issue when upgrading to latest JSON-LD (3.2.0) in my jsonld-signatures fork. It was quite easy to resolve. @graph property did exist on my case, but it was failing to deconstruct the object properly. Wasn't exactly sure why so I made these modifications:
https://github.com/docknetwork/jsonld-signatures/blob/master/lib/suites/LinkedDataSignature.js#L258
https://github.com/docknetwork/jsonld-signatures/blob/master/lib/purposes/ControllerProofPurpose.js#L100

Now everything works as it did before, no more iterator errors. Let me know if this doesn't look correct. I can always open a PR here if its correct and that helps.

**Comment by mattcollier** [comment]
There is an open PR for upgrading jsonld: https://github.com/digitalbazaar/jsonld-signatures/pull/122

**Comment by dmitrizagidulin** [comment]
Implemented in PR #122, closing.
